packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -118,6 +120,24 @@ library LibProposing {
             );
         }
 
+        {
+            IProofVerifier proofVerifier = IProofVerifier(
+                resolver.resolve("proof_verifier", false)
+            );
+            bool verified = proofVerifier.verifyZKP({
```

Comment:
Hopefully we can postpone this proof verification here because quite expensive:
- Only if this is invalid, a "prover" can run this proof verification in the smart contract and prove that the block is invalid.
- Otherwise this proof is part of the main proof and is proven together in the aggregation circuit off-chain with the proofs of the other sub-circuits.

Reply:
Good point, will change it later.